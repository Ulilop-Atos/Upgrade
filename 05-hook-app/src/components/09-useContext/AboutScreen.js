import { useContext } from "react";
import { UserContext } from "./UserContext";

export const AboutScreen = () => {
  const { user, setUser } = useContext(UserContext);

  const handleClick = () => {
    setUser({});
  };

  return (
    <>
      <h1>AboutScreen</h1>
      <hr />
      <pre className="container">
        {JSON.stringify(user,null,2)}
      </pre>

      <button
        className="btn btn-primary"
        onClick={handleClick}
      >
        Logout
      </button>
    </>
  )
}

export default AboutScreen;