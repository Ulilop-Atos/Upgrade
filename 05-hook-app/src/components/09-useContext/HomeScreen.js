import { useContext } from "react";
import { UserContext } from "./UserContext";

export const HomeScreen = () => {

  const { user } = useContext(UserContext);

  return (
    <>
      <h1>HomeScreen</h1>
      <hr />
      <pre className="container">
        {JSON.stringify(user,null,2)}
      </pre>
    </>
  )
}

export default HomeScreen;