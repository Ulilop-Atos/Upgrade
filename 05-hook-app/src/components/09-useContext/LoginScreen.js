import { useContext } from "react";
import { UserContext } from "./UserContext";

export const LoginScreen = () => {

  const {setUser} = useContext(UserContext);

  const userExample = {
    id: 123,
    name: 'Ferdinand',
    email: 'ferdinand@mail.com',
  };

  return (
    <>
      <h1>LoginScreen</h1>
      <hr />
      <button
        className="btn btn-primary"
        onClick={() => setUser(userExample)}
      >
        Login
      </button>
    </>
  )
}

export default LoginScreen;