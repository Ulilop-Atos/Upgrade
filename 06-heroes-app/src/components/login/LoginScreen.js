import { useContext } from "react";
import { useNavigate } from "react-router-dom";
import { AuthContext } from "../../auth/AuthContext";
import { types } from "../../types";

export const LoginScreen = () => {
  
  const {dispatch} = useContext(AuthContext);
  const navigate = useNavigate();  
  
  let name = 'Log Username';
  let mail = 'temp@mail.com';
  
  const handleLogin = () => {
    
    const action = {
      type: types.login,
      payload: {
        name,
        mail,
      },
    }
    
    dispatch(action);
  
    const lastPath = localStorage.getItem('lastPath') || '/';
    
    navigate(lastPath , {
      replace: true
    });
  }

  return (
    <div className="container mt-5">
      <h1>Login</h1>
      <hr />

      <button
        className="btn btn-primary"
        onClick={handleLogin}
      >
        Login
      </button>
    </div>
  )
}
