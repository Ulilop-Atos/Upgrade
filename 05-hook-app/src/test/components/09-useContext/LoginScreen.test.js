import { mount } from "enzyme";
import LoginScreen from "../../../components/09-useContext/LoginScreen";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe('Tests for <LoginScreen />', () => {

  const user = {
    id: 0, 
    name: 'Test Name',
    email: 'test@mail.com',
  }

  const setUser = jest.fn();

  const wrapper = mount(
    <UserContext.Provider value={{user, setUser}}>
      <LoginScreen />
    </UserContext.Provider>
  );

  test('should display properly', () => { 
    expect(wrapper).toMatchSnapshot();
  });

  test('should login properly', () => { 

    const userExample = {
      id: 123,
      name: 'Ferdinand',
      email: 'ferdinand@mail.com',
    };

    wrapper.find('button').simulate('click');
    expect(setUser).toHaveBeenCalledTimes(1);
    expect(setUser).toHaveBeenCalledWith(userExample);
  });
})