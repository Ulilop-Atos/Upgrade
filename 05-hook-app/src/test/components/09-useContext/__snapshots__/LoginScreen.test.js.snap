// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <LoginScreen /> should display properly 1`] = `
Array [
  <h1>
    LoginScreen
  </h1>,
  <hr />,
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>,
]
`;
